src/wp-includes/html-api/class-wp-html-open-elements.php

```diff
@@ -264,43 +339,88 @@ public function has_element_in_list_item_scope( string $tag_name ): bool {
 		return $this->has_element_in_specific_scope(
 			$tag_name,
 			array(
-				// There are more elements that belong here which aren't currently supported.
+				'APPLET',
+				'BUTTON',
+				'CAPTION',
+				'HTML',
+				'TABLE',
+				'TD',
+				'TH',
+				'MARQUEE',
+				'OBJECT',
 				'OL',
+				'TEMPLATE',
 				'UL',
+				// @todo: Support SVG and MathML nodes when support for foreign content is added.
 			)
 		);
 	}
 
 	/**
 	 * Returns whether a particular element is in button scope.
 	 *
+	 * > The stack of open elements is said to have a particular element
+	 * > in button scope when it has that element in the specific scope
+	 * > consisting of the following element types:
+	 * >
+	 * >   - All the element types listed above for the has an element in scope algorithm.
+	 * >   - button in the HTML namespace
+	 *
 	 * @since 6.4.0
+	 * @since 6.7.0 Full implementation.
```

Comment:
Foreign content is not covered yet.
```suggestion
	 * @since 6.7.0 Implementation covering all HTML elements.
```

Reply:
good thought: clarified in 0c9c40f26c8c8749cadf1563cb51357d12852898